src/soorgeon/split.py

```diff
@@ -87,7 +96,23 @@ def _get_h2_header(md):
     found = None
 
     for line in lines:
-        match = re.search(r'\s*##\s+(.+)', line)
+        match = re.search(r'^\s*##\s+(.+)', line)
+
+        if match:
+            found = _sanitize_name(match.group(1))
+
+            break
+
+    return found
+
+
+def _get_h1_header(md):
```

Comment:
given that `_get_h1_header` and `_get_h2_header`, we can create a "factory" that generates both, something like this:

```
def _get_header_factory(regex):
    def _get_header(md):
         # put the body of the function here, and pass regex to re.search
    
    return _get_header


_get_h1_header  = _get_header_factory(regex_for_h1)
_get_h2_header  = _get_header_factory(regex_for_h2)
```